lib/membrane_h264_ffmpeg/parser.ex

```diff
@@ -127,14 +128,26 @@ defmodule Membrane.H264.FFmpeg.Parser do
   end
 
   @impl true
-  def handle_process(:input, buffer, ctx, state) do
-    {payload, state} =
-      if is_nil(state.frame_prefix) or carries_parameters_in_band?(buffer.payload) do
-        {buffer.payload, state}
-      else
-        {state.frame_prefix <> buffer.payload, %{state | frame_prefix: nil}}
-      end
+  def handle_process(:input, buffer, _ctx, %{frame_prefix: <<>>} = state) do
+    do_process(buffer, state)
+  end
 
+  @impl true
+  def handle_process(:input, buffer, _ctx, state) when state.frame_prefix != <<>> do
```

Comment:
The PR title mentions 40 bytes - is that size always needed to properly parse the first NALUs? Can't we just wait and redemand until `byte_size(buffer) + byte_size(state.partial_frame) > 40`? without handling `:not_enough_data` below?

Reply:
I probably should have mentioned that in the comment, but I found out that sometimes it is possible that SEI arrives before SPS and PPS, so we would have to wait for 40 bytes + whatever SEI size is, and it's quite large. Moreover, if we ever find other NALu that can appear before video frame... well this approach wouldn't scale.

Now we are buffering until: we detect that SPS and PPS are in-band and they take priority over those supplied in caps, or we detect a video frame, in which case we inject sps and pps from the caps.